crates/scmi/src/devices/iio.rs

```diff
@@ -0,0 +1,843 @@
+            custom_exponent = max(-16 - unit_exponent, custom_exponent);
+            debug!(
+                "Setting custom scaling coefficient for {}: {}",
+                &path, custom_exponent
+            );
+        }
+        custom_exponent
+    }
+
+    fn add_axis(&mut self, axes: &mut Vec<Axis>, path: &str) {
+        let unit_exponent = UNIT_MAPPING
+            .iter()
+            .find(|mapping| mapping.channel == self.channel)
+            .map_or(0, |mapping| mapping.unit_exponent);
+        // To get meaningful integer values, we must adjust exponent to
+        // the provided scale if any.
+        let custom_exponent = self.custom_exponent(path, unit_exponent);
+        axes.push(Axis {
+            path: String::from(path),
+            unit_exponent,
+            custom_exponent,
+        });
+    }
+
+    fn register_iio_file(&mut self, file: fs::DirEntry, axes: &mut Vec<Axis>) {
+        let channel = self.channel.clone();
+        let raw_file_name = file.file_name();
+        let raw_suffix = "_raw";
+        let file_name = raw_file_name.to_str().unwrap_or_default();
+        if file_name == "name" {
+            self.set_sensor_name_from_file(&file.path());
+        } else if file_name.starts_with(&channel) && file_name.ends_with(raw_suffix) {
+            let infix = &file_name[channel.len()..file_name.len() - raw_suffix.len()];
+            let infix_len = infix.len();
+            if infix_len == 0 || (infix_len == 2 && infix.starts_with('_')) {
+                let raw_axis_path = Path::new(&self.path)
+                    .join(Path::new(&file_name))
+                    .to_str()
+                    .unwrap()
+                    .to_string();
+                let axis_path = raw_axis_path.strip_suffix(raw_suffix).unwrap();
+                self.add_axis(axes, axis_path);
+                if infix_len > 0 {
+                    self.scalar = false;
+                }
+            }
+        }
+    }
+
+    fn read_axis_file<T: FromStr>(
+        &self,
+        path: &str,
+        name: &str,
+    ) -> Result<Option<T>, ScmiDeviceError> {
+        for value_path in [
+            Path::new(&(String::from(path) + "_" + name)),
+            &Path::new(&path).parent().unwrap().join(name),
+        ]
+        .iter()
+        {
+            let value: Option<T> = read_number_from_file(value_path)?;
+            if value.is_some() {
+                return Ok(value);
+            }
+        }
+        Ok(None)
+    }
+
+    fn read_axis_offset(&self, path: &str) -> Result<Option<i64>, ScmiDeviceError> {
+        self.read_axis_file(path, "offset")
+    }
+
+    fn read_axis_scale(&self, path: &str) -> Result<Option<f64>, ScmiDeviceError> {
+        self.read_axis_file(path, "scale")
+    }
+
+    fn read_axis(&self, axis: &Axis) -> Result<i64, ScmiDeviceError> {
+        let mut value: i64 =
+            read_number_from_file(Path::new(&(String::from(&axis.path) + "_raw")))?.unwrap();
```

Comment:
```suggestion
            read_number_from_file(Path::from(format!("{}_raw", axis.path))?.unwrap();
```

Reply:
This piece of code has changed a bit, so no longer applicable.